src/dos/dos_mscdex.cpp

```diff
@@ -304,6 +310,14 @@ int CMscdex::AddDrive(Bit16u _drive, char* physicalPath, Bit8u& subUnit)
 	int result = 0;
 	// Get Mounttype and init needed cdrom interface
 	switch (MSCDEX_GetMountType(physicalPath)) {
+	case MountType::PHYSICAL:
+		LOG(LOG_MISC, LOG_NORMAL)("MSCDEX: Mounting physical cdrom: %s", physicalPath);
+#if defined (LINUX)
+		cdrom[numDrives] = new CDROM_Interface_Ioctl();
+#else
+		cdrom[numDrives] = new CDROM_Interface_SDL();
+#endif
+		break;
```

Comment:
This is strange - so Windows and macOS are getting `CDROM_Interface_SDL` but Linux needs `CDROM_Interface_Ioctl`? I thought the idea was that there OS-specific code filled in different content all behind a common class? (like `CDROM_Interface_SDL`)?

Reply:
The CDROM_Interface_Ioctl interface is implemented on top of the common CDROM_Interface_SDL interface, and vanilla DOSBox implemented few Ioctl functions overriding the same-named functions in the common CDROM_Interface_SDL interface to make direct sector access work in Linux, and this PR adds the code as is.